packages/medusa/src/api/routes/admin/batch/create-batch-job.ts

```diff
@@ -0,0 +1,49 @@
+import { IsJSON, IsObject, IsString } from "class-validator"
+import { BatchJobService } from "../../../../services"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [post] /batch
+ * operationId: "PostBatches"
+ * summary: "Create a Batch Job"
+ * description: "Creates a Batch Job."
+ * x-authenticated: true
+ * parameters:
+ *   - (body) type=* {string} The type of batch job to start.
+ *   - (body) context=* {string} Additional infomration regarding the batch to be used for processing.
+ * tags:
+ *   - Customer
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *            batch_job:
+ *              $ref: "#/components/schemas/batch_job"
+ */
+export default async (req, res) => {
+  const validated = await validator(AdminPostBatchesReq, req.body)
+
+  const userId: string = req.user.id || req.user.userId
+
+  const toCreate = {
+    created_by: userId,
+    ...validated,
+  }
+
+  const batchJobService: BatchJobService = req.scope.resolve("batchJobService")
+
+  const batch_job = await batchJobService.create(toCreate)
+
+  res.status(200).json({ batch_job })
+}
+
+export class AdminPostBatchesReq {
+  @IsString()
+  type: string
+
+  @IsJSON()
+  context: Record<string, unknown>
+}
```

Comment:
issue: Add `dry_run` as a separate prop., since context will be sent to validation in [strategies](https://github.com/medusajs/medusa/issues/1264) so we don't have to filter it out.

Reply:
Added, should we push this to the context before saving or should we add the field to the database?